android/src/app/organicmaps/DownloadResourcesLegacyActivity.java

```diff
@@ -347,7 +356,7 @@ public void showMap()
     final Intent intent = new Intent(this, MwmActivity.class);
 
     // Disable animation because MwmActivity should appear exactly over this one
-    intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION | Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
This reverts the fix for the crash https://github.com/organicmaps/organicmaps/issues/4240

Should it be really reverted? Why? A detailed comment in the code here would be helpful for anyone who tries to fix crashes again using the same approach.
And if it introduces crashes again, then the crash issue should be reopened. And it also has high priority, like a release blocker.

Reply:
> 

`FLAG_ACTIVITY_NEW_TASK` creates a new task and this breaks propagation of the activity result from MwmActivity to DownloadLegacyActivity -> SplashActivity.

Reply:
Did you check if #4240 reproduces again on this branch?

Reply:
No, #4240 was fixed by #4656.